site/en/r2/guide/keras/training_and_evaluation.ipynb

```diff
@@ -365,6 +394,89 @@
       "source": [
         "#### Writing custom losses and metrics\n",
         "\n",
+        "There are 2 ways to provide custom losses outside of the keras API. One is to make and pass a function that accepts inputs `y_true` and `y_pred`. An example of how to do this is shown below in a loss function that computes the average distance between the real data and the predictions:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "7m4P75YEFa8x",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "def dumb_loss_function(y_true, y_pred):\n",
+        "    return tf.math.reduce_mean(y_true-y_pred)\n",
+        "\n",
+        "model.compile(optimizer=keras.optimizers.RMSprop(learning_rate=1e-3),\n",
+        "              loss=dumb_loss_function)\n",
+        "model.fit(x_train, y_train,\n",
+        "          batch_size=64,\n",
+        "          epochs=3)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Q92L8EvxFa82",
+        "colab_type": "text"
+      },
+      "source": [
+        "If you need a loss function that takes in parameters beside `y_true` and `y_pred` and isn't part of the API, you can create them by subclassing the `Loss` class. You will need to implement 2 methods:\n",
+        "\n",
+        "- `__init__(self)`, in which you will accept parameters to pass during the call of your loss function\n",
+        "- `call(self, y_true, y_pred)`, which uses the targets `y_true` and the model predictions `y_pred` to compute the model's loss\n",
+        "\n",
+        "Parameters passed into `__init__()` can be used during `call()` when calculating loss.\n",
+        "\n",
+        "Here is an example showing how to implement a `WeightedCrossEntropy` loss function which calculates a `BinaryCrossEntropy` loss where the loss of a certain class or the whole function can be modified by a scalar."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "Ch5kZdUOFa83",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "source": [
+        "class WeightedBinaryCrossEntropy(keras.losses.Loss):\n",
+        "    \n",
```

Comment:
Add comments about the arguments used.

Args:
  pos_weight: ...
  weight: ...